### PR TITLE
feat(profiling): Introduce profile timeseries query builder

### PR DESCRIPTION
**Pull request by Zylphrex**
Analogous to #40557, this introduces a new query builder to be able to return timeseries data.

### CODE REVIEW

File: src/sentry/api/endpoints/organization_events_stats.py [hunk 1]
```diff
@@ -154,11 +154,19 @@ def get(self, request: Request, organization: Organization) -> Response:
                     )
                 )
             )
+
+            use_profiles = features.has(
```
Comment (wmak) [inline src/sentry/api/endpoints/organization_events_stats.py:158]: Could you add this to `self.get_features` so it gets batched? (lol sorry I don't have this on the /events/ api)

**Review by wmak — APPROVED** [review]
lgtm

File: src/sentry/api/endpoints/organization_events_stats.py (hunk 1 above)
Comment (Zylphrex) [reply src/sentry/api/endpoints/organization_events_stats.py:158]: this isnt on flagr so there's no benefit to doing this

File: src/sentry/api/endpoints/organization_events_stats.py [hunk 2]
```diff
@@ -154,11 +154,19 @@ def get(self, request: Request, organization: Organization) -> Response:
...
+            )
+
             performance_dry_run_mep = batch_features.get(
                 "organizations:performance-dry-run-mep", False
             )
 
-            dataset = self.get_dataset(request) if use_metrics else discover
+            use_custom_dataset = use_metrics or use_profiles
+            dataset = self.get_dataset(request) if use_custom_dataset else discover
             metrics_enhanced = dataset != discover
```
Comment (wmak) [inline src/sentry/api/endpoints/organization_events_stats.py:170]: doesn't this need to be dataset in {metrics, metrics_enhanced}?

**Pull request merged**